service/raft/group0_state_machine.cc

```diff
@@ -83,12 +123,98 @@ future<> group0_state_id_handler::advertise_state_id(utils::UUID state_id) {
     return _gossiper.get().add_local_application_state(gms::application_state::GROUP0_TOMBSTONE_GC_TIME, gms::versioned_value::gc_clock(gc_time));
 }
 
+future<> group0_state_id_handler::on_change(gms::inet_address endpoint, const gms::application_state_map& states, gms::permit_id pid) {
+    return on_application_state_change(
+        endpoint,
+        states,
+        gms::application_state::GROUP0_TOMBSTONE_GC_TIME,
+        pid,
+        [this](gms::inet_address endpoint, const gms::versioned_value& value, gms::permit_id) {
+            const auto gc_time = gc_clock::from_time_t(std::stol(value.value()));
+
+            slogger.debug("Tombstone GC time received: {}, endpoint count: {}", gc_time, _gossiper.get().get_endpoints().size());
+
+            _gc_times[endpoint] = gc_time;
```

Comment:
The `_gc_times` map should be host ID based, not IP based.

Yes the gossiper provides us with IP but that is due to legacy reasons. We are in the process of migrating to be fully host ID based everywhere

so basically the first thing we should do when we get a gossiper notification is to get the host ID of the node which is sending it. It is directly available from the HOST_ID application state.

Reply:
changed the `_gc_times` to be host based.